calico_kubernetes/calico_kubernetes.py

```diff
@@ -724,6 +731,73 @@ def _log_interfaces(namespace):
         # Don't exit if we hit an error logging out the interfaces.
         logger.exception("Ignoring error logging interfaces")
 
+
+def load_config():
+    """
+    Loads configuration for the plugin - returns a dictionary.
+
+    Looks first in environment, then in local config file.
+    """
+    # First, read the config file and get defaults.
+    config = read_config_file()
+
+    # Get config from environment, if defined.
+    for var in ENVIRONMENT_VARS:
+        config[var] = os.environ.get(var, config[var])
+
+    # ETCD_AUTHORITY is handled slightly differently - we need to set it in the
+    # environment so that libcalico works correctly.
+    if not ETCD_AUTHORITY_VAR in os.environ:
+        logger.debug("Use env variable: %s=%s",
+                     ETCD_AUTHORITY_VAR,
+                     config[ETCD_AUTHORITY_VAR])
+        os.environ[ETCD_AUTHORITY_VAR] = config[ETCD_AUTHORITY_VAR]
+
+    # Ensure case is correct.
+    config[LOG_LEVEL_VAR] = config[LOG_LEVEL_VAR].upper()
+
+    return config
+
+
+def read_config_file():
+    """
+    Reads the config file on disk and returns configuration dictionary.
+    """
+    # Get the current directory and find path to config file.
+    executable = sys.argv[0]
+    cur_dir = os.path.dirname(executable)
+    config_file = os.path.join(cur_dir, CONFIG_FILENAME)
+
+    # Check that the file exists.
+    if not os.path.isfile(config_file):
+        sys.exit("Cannot find configuration file: %s" % config_file)
+
+    # Create dictionary of default values.
+    defaults = {
+        ETCD_AUTHORITY_VAR: "127.0.0.1:2379",
+        CALICO_IPAM_VAR: "true",
+        KUBE_API_ROOT_VAR: "http://kubernetes-master:8080/api/v1",
+        DEFAULT_POLICY_VAR: "allow",
+        KUBE_AUTH_TOKEN_VAR: None,
+        LOG_LEVEL_VAR: "INFO",
+    }
+    config = {}
+
+    # Read the config file.
+    parser = ConfigParser.ConfigParser(defaults)
+    parser.read(config_file)
+
+    # Make sure the config section exists
+    if not "config" in parser.sections():
+        sys.exit("No [config] section in file %s" % config_file)
+
+    # Get any values from the configuration file and populate dictionary.
+    for var in ENVIRONMENT_VARS:
+        config[var] = parser.get("config", var)
```

Comment:
Nope, its a bit weird.

I pass in default values when I create the ConfigParser above.  This call is getting the value of `var` from the section `config`, so I think this is right.